std/src/main/scala/cats/std/map.scala

```diff
@@ -0,0 +1,38 @@
+package cats
+package std
+
+
+trait MapInstances {
+  type MapWithK[A] = Map[A, _]
```

Comment:
Thanks for the work on this, @bthuillier!

I think Map instances will need to be defined by functions (`def`s) that constrain the left side as opposed to `val`s. I imagine it will need to end up looking something like the [`Or` instances](https://github.com/non/cats/blob/master/data/src/main/scala/cats/data/Or.scala#L393). Admittedly I haven't tried this out, so I could be off about the right approach.